Mathlib/Analysis/Complex/Hadamard.lean

```diff
@@ -0,0 +1,487 @@
+
+- `Complex.HadamardThreeLines.closedStrip` : The vertical strip defined by : `re ⁻¹' Icc a b`
+
+- `Complex.HadamardThreeLines.bddStrip` :
+    The rectangle defined by : `re ⁻¹' Ioo a b ∩ im ⁻¹' Ioo (-T) T`
+
+- `Complex.HadamardThreeLines.closedBddStrip` :
+    The rectangle defined by : `re ⁻¹' Icc a b ∩ im ⁻¹' Icc (-T) T`
+
+- `Complex.HadamardThreeLines.sSupAbsIm` :
+    The supremum function on vertical lines defined by : `sSup {|f(z)| : z.re = x}`
+
+- `Complex.HadamardThreeLines.invInterpStrip` :
+    Inverse of the interpolation between the `sSupAbsIm` on the edges of the strip.
+
+- `Complex.HadamardThreeLines.F` :
+    Function defined by `f` times `invInterpStrip`. Convenient form for proofs.
+
+- `Complex.HadamardThreeLines.F'` :
+    'Easy' function converging to `F`.
+
+- `Complex.HadamardThreeLines.cocompactStrip` :
+    The filter along the vertical strip `re ⁻¹' Ioo a b` and `|z.im| atTop`.
+
+## References
+
+This proof follows loosely the proof found on the wiki page :
+[wiki](https://en.wikipedia.org/wiki/Hadamard_three-lines_theorem)
+
+-/
+
+
+open Set Filter Function Complex
+
+namespace Complex
+namespace HadamardThreeLines
+variable (z : ℂ)
+
+/-- The strip in the complex plane containing all `z ∈ ℂ` such that `z.re ∈ Ioo a b`. -/
+def strip (a : ℝ) (b : ℝ) : Set ℂ := re ⁻¹' Ioo a b
+
+/-- The strip in the complex plane containing all `z ∈ ℂ` such that `z.re ∈ Icc a b`. -/
+def closedStrip (a : ℝ) (b : ℝ) : Set ℂ := re ⁻¹' Icc a b
+
+/-- The strip in the complex plane containing all `z ∈ ℂ` such that `z.re ∈ Ioo a b` and
+`z.im ∈ Ioo (-T) T`. -/
+def bddStrip (a : ℝ) (b : ℝ) (T : ℝ) : Set ℂ :=  re ⁻¹' Ioo a b ∩ im ⁻¹' Ioo (-T) T
+
+/-- The strip in the complex plane containing all `z ∈ ℂ` such that `z.re ∈ Icc a b` and
+`z.im ∈ Icc (-T) T`. -/
+def closedBddStrip (a : ℝ) (b : ℝ) (T : ℝ) : Set ℂ :=  re ⁻¹' Icc a b ∩ im ⁻¹' Icc (-T) T
+
+/-- The filter along the vertical strip `re ⁻¹' Ioo a b` and `|z.im| atTop` -/
+def cocompactStrip (a b : ℝ) : Filter ℂ :=
+  comap im (cocompact ℝ) ⊓ (comap re (principal (Icc a b)))
+
+/-- The supremum of the absolute value of `f` on imaginary lines. (Fixed real part) -/
+noncomputable def sSupAbsIm (f : ℂ → ℂ) (x : ℝ) : ℝ  :=  sSup ((abs ∘ f) '' (re ⁻¹' {x}))
+
+
+/-- The inverse of the interpolation of `sSupAbsIm` on the two boundaries. -/
+noncomputable def invInterpStrip (f : ℂ → ℂ) (z : ℂ) : ℂ :=
+  if (sSupAbsIm f (0 : ℝ)) = (0 : ℝ) ∨ (sSupAbsIm f (1 : ℝ)) = (0 : ℝ)
+    then 0
+    else (sSupAbsIm f 0)^(z-1) * (sSupAbsIm f 1)^(-z)
+
+/-- A function useful for the proofs steps. We will aim to show that it is bounded by 1. -/
+private noncomputable def F (f : ℂ → ℂ) := fun z ↦ f z • invInterpStrip f z
+
+/-- Similar to `F` only 'easier'. Useful for proof steps. -/
+private noncomputable def F' (n : ℕ) (f : ℂ → ℂ) := fun z ↦ F f z • exp ((z^2-1) * (n : ℝ)⁻¹)
+
+
+-- Small lemma : Sup of abs is nonneg
```

Comment:
The comment introduced with `--` will show up nowhere. If instead you use `/-- Sup of abs is nonneg -/`, then it will show up in the tooltip over lemma uses in vscode. Maybe make the docstring a little bit less cryptic, though (abs of what?)

Same thing below.

Reply:
Fixed it.